server/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -379,4 +383,36 @@ public Result destroyEligibility(long programId, long blockDefinitionId) {
                 programId, blockDefinitionId))
         .flashing(FlashKey.SUCCESS, "Removed the eligibility condition for this screen.");
   }
+
+  /** POST endpoint for updating eligibility message. */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result updateEligibilityMessage(Request request, long programId, long blockDefinitionId) {
+    requestChecker.throwIfProgramNotDraft(programId);
+
+    Form<BlockEligibilityMessageForm> EligibilityMsgform =
+        formFactory.form(BlockEligibilityMessageForm.class).bindFromRequest(request);
+    String newMessage = EligibilityMsgform.get().getEligibilityMessage();
+    String toastMessage;
+    String toastType;
+
+    try {
+      programService.setBlockEligibilityMessage(
+          programId, blockDefinitionId, Optional.of(LocalizedStrings.of(Locale.US, newMessage)));
+
+      toastType = "success";
```

Comment:
Can we use the value from the enum?
https://github.com/civiform/civiform/blob/441995d2150ed4fc6f0b1ce46f2adf388e34edb9/server/app/views/components/ToastMessage.java#L18